Added "Accept terms of use" checkbox

Laurenzak opened the pull request:
Added two options in the backend to show an "accept terms of use" checkbox on the frontend.
Files affected:

- /components/com_jcomments/tpl/default/style.css: Minimum change to display the form responsive.
- components/com_jcomments/jcomments.php: Redirect "index.php" to homepage
- administrator/component/com_jcomments/version.php: JComments version and Github link
- Backend languages: "administrator/language/en-GB/en-GB.com_jcomments.ini" and "administrator/language/ru-RU/ru-RU.com_jcomments.ini".
- Fronted languages: All language files (43 files "xx-XX.com_jcomments.ini") have been updated.
- Backend option: file "administrator/components/com_jcomments/models/forms/settings.xml" and permissions.xml.
- Database options (2 options): file "administrator/components/com_jcomments/install/sql/default.settings.sql".
- Code to show checkbox. File "components/com_jcomments/tpl/default/tpl_form.php".
- Code to create a variable: File "components/com_jcomments/jcomments.php"
- Code to check if the checkbox has been activated in the backend and if the user has checked the checkbox. File "components/com_jcomments/jcomments.ajax.php"and saveComment function.

 Fixed error in recaptcha v2. Deleted RECAPTCHA item in JComments CAPTCHA options. ERROR fixed: disable recaptcha v2 from Joomla plugins and then select RECAPTCHA option in JComments. File administrator/models/forms/settings.xml


Important: The files in your repository are not the same as those in your "com_jcomments_v3.0.7.5.zip" file. Three files affected: /site/tpl/default/tpl_form.php, site/tpl/default/tpl_index.php and site\js\jcomments-v2.3.js.

Laurenzak committed:
Checkbox "Terms of use" added

Added "Accept terms of use" checkbox option.

Laurenzak committed:
Delete default.settings.sql.bak

Laurenzak committed:
Delete ru-RU.com_jcomments.ini.bak

Laurenzak committed:
Delete en-GB.com_jcomments.ini.bak

Laurenzak committed:
Deleted .bak

Laurenzak committed:
Merge branch 'Develop' of https://github.com/Laurenzak/Jcomments-Recaptcha2 into Develop

Laurenzak committed:
style.css updated

I was using the original version 3.0.7, now it has been mixed with 3.0.7.5 version (from exstreme)

Laurenzak commented on administrator/models/forms/settings.xml:
```diff
@@ -403,7 +415,6 @@
                 label="AP_FORM_FIELD_CAPTCHA"
                 description="">
             <option value="kcaptcha">KCAPTCHA</option>
-			<option value="recaptcha">RECAPTCHA</option>
```
This code creates an item in the JComments captcha dropdown (backend > layout > CAPTCHA). In the official JComments 3.0.7 version, it does not appear. Because it was an option to use recapcha v2 in JComments. But now it can throw an error.
Steps to reproduce the error: disable recaptcha v2 from Joomla plugins and then select "RECAPTCHA" option in JComments.

exstreme commented:
Greetings!
Why you change JFactory::getDocument()->addScriptDeclaration($script); to <script></script> ?
It can broke some plugins work with speed optimization.

exstreme commented:
And Recaptcha Invisible is v2, but not v3

Laurenzak commented:
> Greetings!
> Why you change JFactory::getDocument()->addScriptDeclaration($script); to <script></script> ?
> It can broke some plugins work with speed optimization.

I have made a mistake. I was using your version 3.0.7.5 from your file "Jcomments-Recaptcha2/com_jcomments_v3.0.7.5.zip" but I noticed that it is outdated. I'm going to fix everything and update the files.

Thank you very much for your time

Laurenzak commented:
> And Recaptcha Invisible is v2, but not v3

Ok. I'm going to change this:
"JComments v3.0.7.X is a modified version of the original Jcomments v3.0.7. (https://github.com/exstreme/Jcomments-Recaptcha2) Link to Github project.
PHP 7.2 compatible
Tested on Joomla v3.9.24
Recaptcha v2 and v3 (invisible)
Accept terms of use' checkbox
And more..."
to this:
"JComments v3.0.7.X is a modified version of the original Jcomments v3.0.7. (https://github.com/exstreme/Jcomments-Recaptcha2) Link to Github project.
PHP 7.2 compatible
Tested on Joomla v3.9.24
Recaptcha v2
Accept terms of use' checkbox
And more..."

Any other change, write it. Joomla uses an "invisible" plugin. This plugin was designed for v2 invisible but on my site I use this plugin with recatpcha v3. Official Joomla doc says: "Choose the type of reCAPTCHA you would like to use. Prefer the reCAPTCHA V3 type." But I don't know if JComments is able to use reCAPTCHA v3.

Laurenzak committed:
Updated from exstreme repository

I was using version 3.0.7.5, but it is outdated. It is now updated to the latest version from the exstreme repository.

Laurenzak committed:
Updated from exstreme repository

Updated with the same spaces

Laurenzak commented:
Updated with the same original code spaces to correctly see the differences.
